src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ThrowHelper.cs

```diff
@@ -68,6 +68,7 @@ static Exception GetExceptionInternal(int status, string? message)
             if (status == QUIC_STATUS_ADDRESS_IN_USE) return new QuicException(QuicError.AddressInUse, null, SR.net_quic_address_in_use);
             if (status == QUIC_STATUS_UNREACHABLE) return new QuicException(QuicError.HostUnreachable, null, SR.net_quic_host_unreachable);
             if (status == QUIC_STATUS_CONNECTION_REFUSED) return new QuicException(QuicError.ConnectionRefused, null, SR.net_quic_connection_refused);
+            if (status == QUIC_STATUS_CONNECTION_TIMEOUT) return new QuicException(QuicError.ConnectionTimeout, null, SR.net_quic_timeout);
```

Comment:
This change looks fine, but subsequently I think this method would be cleaner if implemented as a switch.

Reply:
the QUIC_STATUS_* members are not compile-time constants, unfortunately

Reply:
> they are get-only properties with if-else per platform (they come from autogenerated .cs files from MsQuic)

😮

Reply:
Do we know why msquic defines different values for these constants per platform?  That sounds like a very leaky abstraction.

Reply:
That's misguided from my perspective, but ok.

Reply:
See
https://github.com/microsoft/msquic/blob/main/src/inc/msquic_winuser.h#L90
https://github.com/microsoft/msquic/blob/main/src/inc/msquic_posix.h#L104

Reply:
I think it's a design flaw that the library is surfacing platform-specific details like this... it doesn't need to in a case like this, and is just pushing the pain upwards to consumers of the library.

Reply:
> I think it's a design flaw that the library is surfacing platform-specific details like this... it doesn't need to in a case like this, and is just pushing the pain upwards to consumers of the library.

I am not thrilled about it either, but at least now they give us autogeneated stubs which contain the right numbers so we can be reasonably sure that we did not typo it.